crowbar_framework/app/models/api/upgrade.rb

```diff
@@ -23,6 +23,13 @@ def status
         ::Crowbar::UpgradeStatus.new.progress
       end
 
+      def node_status
+        {
+          upgraded: ::Node.all.reject(&:admin?).select(&:ready_after_upgrade?).map(&:name),
+          not_upgraded: ::Node.find("state:crowbar_upgrade").map(&:name)
```

Comment:
Hm, you are right that before `prepare` (heck, even before `nodes`) it does not make sense.
Maybe we could just check for the current step here?